File: platform_src/android/app/src/main/java/org/elastos/trinity/runtime/IntentFilter.java
package org.elastos.trinity.runtime;

public class IntentFilter {
    public String packageId = null;
    public String action;
    public String startupMode;
    public String serviceName;

    IntentFilter(String action, String startupMode, String serviceName) {
        this.action = action;
        this.startupMode = startupMode;
        this.serviceName = serviceName;
    }
}

File: platform_src/electron/main/IntentFilter.ts
import { PrimaryGeneratedColumn, Column, Entity } from 'typeorm';

@Entity()
export class IntentFilter {
    @PrimaryGeneratedColumn() public tid: string;
    @Column() app_id: string;
    @Column({ nullable: true }) public packageId: string = null;
    @Column({ nullable: true }) public action: string;
    @Column({ nullable: true }) public startupMode: string;
    @Column({ nullable: true }) public serviceName: string;

    constructor(action: string, startupMode: string, serviceName: string) {
        this.action = action;
        this.startupMode = startupMode;
        this.serviceName = serviceName;
    }
}